stacker_blueprints/dynamodb.py

```diff
@@ -0,0 +1,152 @@
+from stacker.blueprints.base import Blueprint
+from . import util
+
+from troposphere import (
+    dynamodb2,
+    Ref,
+    GetAtt,
+    Output,
+)
+
+from troposphere.route53 import RecordSetType
+
+def prep_schemata(config):
+    try:
+        schemata = []
+        for schema in config["KeySchema"]:
+            schemata.append(dynamodb2.KeySchema(**schema))
+
+    except IndexError:
+        raise IndexError(
+            "The key schema is required for the creation of a DynamoDB table."
+        )
+
+    return schemata
+
+def prep_projection(config):
+    return dynamodb2.Projection(**config["Projection"])
+
+def prep_throughput(config):
+    try:
+        return dynamodb2.ProvisionedThroughput(
+            **config["ProvisionedThroughput"]
+        )
+    except IndexError:
+        raise IndexError(
+            "ProvisionedThroughput values are required for the creation of " +
+            "a DynamoDB table or index."
+        )
+
+def prep_config(raw_config):
+    prepped_config = {}
+
+    dynamodb_table_properties = [
+        "AttributeDefinitions",
+        "GlobalSecondaryIndexes",
+        "KeySchema",
+        "LocalSecondaryIndexes",
+        "ProvisionedThroughput",
+        "StreamSpecification",
+    ]
+
+    util.check_properties(
+        raw_config,
+        dynamodb_table_properties,
+        "DynamoDB",
+    )
+
+    # AttributeDefinitions are required, so raise an IndexError if this
+    # doesn't work.
+    try:
+        attributes = []
+        for attribute in raw_config["AttributeDefinitions"]:
+            attributes.append(dynamodb2.AttributeDefinition(**attribute))
+
+        prepped_config["AttributeDefinitions"] = attributes
+    except IndexError:
+        raise IndexError(
+            "Attribute definitions are required for the creation of a " +
+            "DynamoDB table."
+        )
+
+    # Global Secondary Index section.
+    if "GlobalSecondaryIndexes" in raw_config:
+        # AWS limits us to 5 GSIs.  Check for that and bail if there's more.
+        if len(raw_config["GlobalSecondaryIndexes"]) > 5:
```

Comment:
I feel like this is too granular of a check. I don't think AWS is likely to change this limit, but if they do, now this template has to update. CloudFormation will throw the error when building the stack anyways, along with any other error, like a value in the KeySchema of the GSI not being present in the original AttributeDefinitions.

I see the value in doing the check, just not sure its necessary